doc/reference/reference_lua/json.rst

```diff
@@ -116,6 +119,179 @@ Below is a list of all ``json`` functions and members.
+
+* 'seq', 'sequence', 'array' - table encoded as an array
+* 'map', 'mapping' - table encoded as a map
+* function - the meta-method called to unpack serializable representation
+  of table, cdata or userdata objects
+
+Serializing 'A' and 'B' with different ``__serialize`` values brings different
+results:
+
+.. code-block:: tarantoolsession
+
+    tarantool> json.encode(setmetatable({'A', 'B'}, { __serialize="seq"}))
+    ---
+    - '["A","B"]'
+    ...
+    tarantool> json.encode(setmetatable({'A', 'B'}, { __serialize="map"}))
+    ---
+    - '{"1":"A","2":"B"}'
+    ...
+    tarantool> json.encode({setmetatable({f1 = 'A', f2 = 'B'}, { __serialize="map"})})
+    ---
+    - '[{"f2":"B","f1":"A"}]'
+    ...
+    tarantool> json.encode({setmetatable({f1 = 'A', f2 = 'B'}, { __serialize="seq"})})
+    ---
+    - '[[]]'
+    ...
+
+.. _json-module_cfg:
+
+.. function:: cfg(table)
+
+    Set values that affect the behavior of :ref:`json.encode <json-encode>`
+    and :ref:`json.decode <json-decode>`.
+
+    The values are all either integers or boolean ``true``/``false``.
+
+    .. container:: table
+
+        .. rst-class:: left-align-column-1
+        .. rst-class:: center-align-column-2
+        .. rst-class:: left-align-column-3
+
+        +---------------------------------+---------+-------------------------------------------+
+        | Option                          | Default | Use                                       |
+        +=================================+=========+===========================================+
+        | ``cfg.encode_max_depth``        |   128   | Max recursion depth for encoding          |
+        +---------------------------------+---------+-------------------------------------------+
+        | ``cfg.encode_deep_as_nil``      |  false  | A flag saying whether to crop tables      |
+        |                                 |         | with too deep nesting level.              |
+        |                                 |         | Not-encoded fields are replaced with      |
+        |                                 |         | one null. If not set, too deep            |
+        |                                 |         | nesting is considered an error.           |
+        +---------------------------------+---------+-------------------------------------------+
+        | ``cfg.encode_invalid_numbers``  |  true   | A flag saying whether to enable encoding  |
+        |                                 |         | of NaN and Inf numbers                    |
+        +---------------------------------+---------+-------------------------------------------+
+        | ``cfg.encode_number_precision`` | 14      | Precision of floating point numbers       |
+        +---------------------------------+---------+-------------------------------------------+
+        | ``cfg.encode_load_metatables``  | true    | A flag saying whether to encode as a map  |
```

Comment:
Unclear: what to encode (metatables)?

Reply:
@Totktonada "to encode a table as a map" is correct?

Reply:
No. It means whether the serializer will follow `__serialize` metatable field. See the [comment](https://github.com/tarantool/tarantool/blob/82cc56316bfefe3a1e7536247878609eeb74f39e/src/lua/utils.h#L225-L234). It affects table, cdata and userdata encoding.

Reply:
@Onvember please add these details to the doc